index.js

```diff
@@ -399,6 +402,35 @@
     function(x) { return $.FiniteNumber.test(x) && Number(x) !== 0; }
   );
 
+  //  Integer :: Type
+  $.Integer = $.NullaryType(
+    'sanctuary-def/Integer',
+    function(x) {
+      return $.ValidNumber.test(x) &&
+        Math.floor(x) === Number(x) &&
+        x >= MIN_SAFE_INTEGER  &&
+        x <= MAX_SAFE_INTEGER;
+    }
+  );
+
+  //  NonZeroInteger :: Type
+  $.NonZeroInteger = $.NullaryType(
+    'sanctuary-def/NonZeroInteger',
+    function(x) { return $.NonZeroValidNumber.test(x) && $.Integer.test(x); }
```

Comment:
I'd prefer to let `$.Integer.test` reject `Infinity`, `-Infinity`, and `NaN`. We could then have a nice symmetry between this predicate and the following two:

```javascript
//  $.NonZeroInteger.test
x => $.Integer.test(x) && x != 0

//  $.PositiveInteger.test
x => $.Integer.test(x) && x > 0

//  $.NegativeInteger.test
x => $.Integer.test(x) && x < 0
```